Shopkeeper/Shopkeeper.Data/ItemRepository.cs

```diff
@@ -4,33 +4,47 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using Microsoft.Extensions.Logging;
 
 namespace Shopkeeper.Data
 {
     public class ItemRepository : IItemRepository
     {
         private readonly IReadOnlyDictionary<string, Item> Items;
         private readonly QuizItems configItems;
+        private readonly ILogger<ItemRepository> logger;
 
-        public ItemRepository(IOptionsMonitor<QuizItems> itemsConfig)
+        public ItemRepository(IOptionsMonitor<QuizItems> itemsConfig, ILogger<ItemRepository> logger)
         {
             configItems = itemsConfig.CurrentValue;
+            this.logger = logger;
+
             Items = LoadJson();
         }
 
         private Dictionary<string, Item> LoadJson()
         {
+            logger.LogInformation("Parsing Item Json");
+
             foreach (var item in configItems.Items.Where(x => x.ComponentNames != null))
             {
-                item.Components = configItems.Items.Where(x => item.ComponentNames.Contains(x.Id));
+                logger.LogDebug("Populating Components for {0}: {1}", item.Name, string.Join(',', item.ComponentNames));
+
+                item.Components = item.ComponentNames
+                    .Select(itemId => configItems.Items.Single(i => i.Id == itemId))
+                    .ToList(); // eager load for easier logging
             }
 
+            logger.LogDebug("{0} Items Found", configItems.Items.Count());
```

Comment:
Consider storing Items as a collection type. Since you store it above as a list anyway and this avoids calling .Count() and instead you can use .Count